htdocs/class/file/file.php

```diff
@@ -439,7 +439,7 @@ public function pwd()
      */
     public function exists()
     {
-        $exists = (file_exists($this->pwd()) && is_file($this->pwd()));
+        $exists = (is_file($this->pwd()) && is_file($this->pwd()));
```

Comment:
This is a perfect example of why these proposed changes are a bad idea.

In virtually all filesystems it is impossible to have both a file and a directory at the same level with the same name. The existence of a directory `/path/data` precludes the existence of a file `/path/data` and vice versa.

Checking exclusively with is_dir() or is_file() can lead to problems where file_exists() could cause a different decision. It is not safe to to assume that a substitution can be made without undesired side effects.

A false return from `is_file('/path/data')` does NOT guarantee that a file `/path/data` can be created.

This should be considered on a case by case basis when implementing new code or when reworking the surrounding code. Those points in the code life cycle will provide the needed context and testing opportunities.